### PR TITLE
[web] Avoid returning `int` from js interop classes.

### PR DESCRIPTION
**Pull request by joshualitt**
While ints can flow to JS on all backends, Dart2Wasm does not currently support returning ints from JS. The reason is because there is really no way to support this in the general case. This CL does not change any public APIs, but rather instead we introduce tiny trampolines to handle the conversion.

### CODE REVIEW

**Comment by Hixie** [comment]
test-exemption: code refactor with no semantic change

**Review by eyebrowsoffire — APPROVED** [review]
LGTM! Thank you!

File: lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart [hunk 1]
```diff
@@ -96,10 +96,18 @@ extension CanvasKitExtension on CanvasKit {
...
+  @JS('UnderlineDecoration')
+  external double get _UnderlineDecoration;
+  @JS('OverlineDecoration')
+  external double get _OverlineDecoration;
+  @JS('LineThroughDecoration')
+  external double get _LineThroughDecoration;
+  int get NoDecoration => _NoDecoration.toInt();
+  int get UnderlineDecoration => _UnderlineDecoration.toInt();
+  int get OverlineDecoration => _OverlineDecoration.toInt();
+  int get LineThroughDecoration => _LineThroughDecoration.toInt();
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart:110]: IMHO, it's better to put each trampoline right under its respective private getter like you did in the rest of the PR:
```suggestion
  @JS('NoDecoration')
  external double get _NoDecoration;
  int get NoDecoration => _NoDecoration.toInt();
  @JS('UnderlineDecoration')
  external double get _UnderlineDecoration;
  int get UnderlineDecoration => _UnderlineDecoration.toInt();
  @JS('OverlineDecoration')
  external double get _OverlineDecoration;
  int get OverlineDecoration => _OverlineDecoration.toInt();
  @JS('LineThroughDecoration')
  external double get _LineThroughDecoration;
  int get LineThroughDecoration => _LineThroughDecoration.toInt();
```

**Review by mdebbar — APPROVED** [review]
Some of these getters might be on a hot path. Do you know how `toInt()` behaves when the value is already an int? Is it a no-op?

Also, it's very easy to introduce int getters without adding a trampoline for them. Is there a way to detect or prevent that from happening? i.e. are the skwasm tests going to fail if an int is returned without doing `toInt()`?

**Comment by joshualitt** [comment]
> Some of these getters might be on a hot path. Do you know how `toInt()` behaves when the value is already an int? Is it a no-op?

Not entirely a noop, but almost. There is a size test, but because nearly all `int`s will be int32, branch prediction will nearly always be correct. Then there is an `int | 0` instruction, but that too should be quite cheap. @yjbanov and I agreed the amount of overhead was negligible. However, it may still make sense to have these APIs return `double` or `num` someday, but really more for code cleanliness than performance.
> 
> Also, it's very easy to introduce int getters without adding a trampoline for them. Is there a way to detect or prevent that from happening? i.e. are the skwasm tests going to fail if an int is returned without doing `toInt()`?

Good point, in the longer term when Wasm tests are running on CI, then these issues will be caught. In addition, I'm going to propose 'compatibility mode' lints for JS interop, and if we have such a set of lints we can add one to catch this issue for users who want their JS interop code to work correctly on all backends.

**Pull request merged**